drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1230 @@
+}
+
+/**
+ * @brief update the measurement range of the X/Y/Z-axis
+ *
+ * @param dev handle to the sensor
+ * @param val value to be set
+ *
+ * @return see @ref i2c_reg_update_byte_dt
+ */
+static inline int tmag5273_attr_set_xyz_meas_range(const struct device *dev,
+						   const struct sensor_value *val)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	struct tmag5273_data *drv_data = dev->data;
+
+	const int16_t range_high = (drv_data->version == TMAG5273_VER_TMAG5273X1)
+					   ? TMAG5273_MEAS_RANGE_HIGH_MT_VER1
+					   : TMAG5273_MEAS_RANGE_HIGH_MT_VER2;
+	const int16_t range_low = (drv_data->version == TMAG5273_VER_TMAG5273X1)
+					  ? TMAG5273_MEAS_RANGE_LOW_MT_VER1
+					  : TMAG5273_MEAS_RANGE_LOW_MT_VER2;
+
+	const uint8_t regdata = (sensor_value_to_double(val) > range_low)
+					? TMAG5273_XYZ_MEAS_RANGE_HIGH
+					: TMAG5273_XYZ_MEAS_RANGE_LOW;
+
+	int retval = i2c_reg_update_byte_dt(&drv_cfg->i2c, TMAG5273_REG_SENSOR_CONFIG_2,
+					    TMAG5273_X_Y_RANGE_MSK | TMAG5273_Z_RANGE_MSK, regdata);
+
+	if (retval == 0) {
+		drv_data->xyz_range =
+			(regdata == TMAG5273_XYZ_MEAS_RANGE_LOW) ? range_low : range_high;
+	}
+
+	return retval;
+}
+
+/**
+ * @brief returns the used measurement range of the X/Y/Z-axis
+ *
+ * @param dev handle to the sensor
+ * @param val return value
+ *
+ * @return \c 0 on success
+ * @return see @ref i2c_reg_read_byte_dt
+ */
+static inline int tmag5273_attr_get_xyz_meas_range(const struct device *dev,
+						   struct sensor_value *val)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	struct tmag5273_data *drv_data = dev->data;
+
+	uint8_t regdata = 0;
+
+	int retval = i2c_reg_read_byte_dt(&drv_cfg->i2c, TMAG5273_REG_SENSOR_CONFIG_2, &regdata);
+
+	if (retval != 0) {
+		return retval;
+	}
+
+	regdata &= (TMAG5273_X_Y_RANGE_MSK | TMAG5273_Z_RANGE_MSK);
+
+	if (regdata == TMAG5273_XYZ_MEAS_RANGE_HIGH) {
+		val->val1 = (drv_data->version == TMAG5273_VER_TMAG5273X1)
+				    ? TMAG5273_MEAS_RANGE_HIGH_MT_VER1
+				    : TMAG5273_MEAS_RANGE_HIGH_MT_VER2;
+	} else {
+		val->val1 = (drv_data->version == TMAG5273_VER_TMAG5273X1)
+				    ? TMAG5273_MEAS_RANGE_LOW_MT_VER1
+				    : TMAG5273_MEAS_RANGE_LOW_MT_VER2;
+	}
+
+	return 0;
+}
+
+/**
+ * set the X/Y/Z angle & magnitude calculation mode
+ *
+ * @param dev handle to the sensor
+ * @param val value to be set
+ *
+ * @return \c -ENOTSUP if unknown value
+ * @return see @ref i2c_reg_update_byte_dt
+ */
+static inline int tmag5273_attr_set_xyz_calc(const struct device *dev,
+					     const struct sensor_value *val)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	uint8_t regval = 0;
+	int retval = -ENOTSUP;
+
+	switch (val->val1) {
+	case TMAG5273_ANGLE_CALC_NONE: {
+		regval = TMAG5273_ANGLE_EN_NONE;
+		retval = 0;
+		break;
+	}
+
+	case TMAG5273_ANGLE_CALC_XY: {
+		if (drv_cfg->x_axis && drv_cfg->y_axis) {
+			regval = TMAG5273_ANGLE_EN_XY;
+			retval = 0;
+		}
+
+		break;
+	}
+
+	case TMAG5273_ANGLE_CALC_YZ: {
```

Comment:
the blocks here look a bit odd, { } aren't needed, you already have a break statement for each case and aren't declaring any variables